apps/06.1Checkers.js

```diff
@@ -3,75 +3,227 @@
-    console.log(string);
-  };
 
-  // Your code here
+
+
+    this.grid = [];
+    // creates an 8x8 array, filled with null values
+    this.createGrid = function() {
+        // loop to create the 8 rows
+        for (var row = 0; row < 8; row++) {
+            this.grid[row] = [];
+            // push in 8 columns of nulls
+            for (var column = 0; column < 8; column++) {
+                this.grid[row].push(null);
+            }
+        }
+    };
+
+    // prints out the board
+    this.viewGrid = function() {
+        // add our column numbers
+        var string = "  0 1 2 3 4 5 6 7\n";
+        for (var row = 0; row < 8; row++) {
+            // we start with our row number in our array
+            var rowOfCheckers = [row];
+            // a loop within a loop
+            for (var column = 0; column < 8; column++) {
+                // if the location is "truthy" (contains a checker piece, in this case)
+                if (this.grid[row][column]) {
+                    // push the symbol of the check in that location into the array
+                    rowOfCheckers.push(this.grid[row][column].symbol);
+                } else {
+                    // just push in a blank space
+                    rowOfCheckers.push(' ');
+                }
+            }
+            // join the rowOfCheckers array to a string, separated by a space
+            string += rowOfCheckers.join(' ');
+            // add a 'new line'
+            string += "\n";
+        }
+        console.log(string);
+    };
+
+    // Your code here
+    this.selectChecker = function(row, column) {
+        return this.grid[row][column];
+    }
 }
+
 function Game() {
 
-  this.board = new Board();
+    this.board = new Board();
 
-  this.start = function() {
-    this.board.createGrid();
-    // Your code here
-  };
+    this.start = function() {
+        this.board.createGrid();
+        // Your code herethis.board.createCheckers()
+        this.board.createCheckers();
+
+    this.moveChecker = function(start, end) {
+        //we need to send in row and column to selectChecker
+        var startrow = start.split('')[0];
+        var startcolumn = start.split('')[1];
+
+        var endrow = end.split('')[0];
+        var endcolumn = end.split('')[1];
+
+        var distance = Math.abs(startrow - endrow);
+
+        if (distance === 2) {
+            var killrow = (parseInt(startrow) + parseInt(endrow)) / 2;
+            var killcolumn = (parseInt(startcolumn) + parseInt(endcolumn)) / 2;
+            var killPosition = [killrow, killcolumn];
+            this.board.killChecker(killPosition);
+        }
+        var checker = this.board.selectChecker(startrow, startcolumn);
+      this.board.grid[endrow][endcolumn] = checker;
+
+        //we are removing the checker from the grid
+        //the grid is in a property called "grid"
+        //we represent removing it by setting the row/column to null
+
+        this.board.grid[startrow][startcolumn] = null;
```

Comment:
ALL TESTS PASSED!

![testspassed](https://media1.giphy.com/media/DOxlkyuDxTrna/200_s.gif)

Your next challenge could be to keep checkers from moving to invalid spots or over existing checkers.